frog/imports/client/App/TopBar.js

```diff
@@ -151,25 +161,35 @@ class TopBarController extends React.Component<
             </Typography>
             <Tabs
               className={classes.tabs}
-              value={this.value}
+              value={this.state.location}
               onChange={this.handleChange}
               variant="fullWidth"
             >
               {this.routes.map(route => (
                 <Tab
-                  key={route.to}
+                  key={route.name}
                   label={route.name}
                   component={Link}
-                  to={route.to + LocalSettings.UrlCoda}
-                  value={route.to}
+                  to={
+                    (route.to.find(to =>
+                      to
+                        .split('/')
+                        .reduce(
+                          (acc, word, idx) =>
+                            acc &&
+                            (!splitPath[idx] || splitPath[idx] === word) ===
+                              word,
+                          true
+                        )
+                    ) || route.to[0]) + LocalSettings.UrlCoda
+                  }
+                  value={route.name}
                 />
               ))}
             </Tabs>
             <h3>
               {LocalSettings.researchLogin ? ' * ' : ''}
-              {Meteor.user().isAnonymous
-                ? 'Anonymous Visitor'
-                : Meteor.user().username}
+              {Meteor.user().username}
```

Comment:
Nit: make sure this change is linked with the PR

Reply:
Updated with develop to undo these changes